onmt/utils/loss.py

```diff
@@ -111,16 +134,15 @@ def sharded_compute_loss(self, batch, output, attns,
           normalization (int) : Loss is divided by this number
 
         Returns:
-            :obj:`onmt.Statistics`: validation loss statistics
+            :obj:`onmt.utils.Statistics`: validation loss statistics
 
         """
-        batch_stats = onmt.Statistics()
+        batch_stats = onmt.utils.Statistics()
         range_ = (cur_trunc, cur_trunc + trunc_size)
         shard_state = self._make_shard_state(batch, output, range_, attns)
-
         for shard in shards(shard_state, shard_size):
             loss, stats = self._compute_loss(batch, **shard)
-            loss.div(float(normalization)).backward()
+            loss.div(float(normalization)).backward(retain_graph=True)
```

Comment:
Can you revert this file to be the same as master? It has some changes that are wrong, and we fixed in master for 0.4.

Reply:
(For instance retain-graph actually is incorrect here and may lead to extra gradients)